projects/golang/go/Makefile

```diff
@@ -118,6 +118,18 @@ tar-golang-sources:
 	mkdir -p $(VERSION_DIRECTORY)/archives/$(GOOS)/$(ARCH_LOWER)
 	cd /tmp/go-extracted/usr/lib/golang && rm -rf ./pkg/obj && tar czf $(VERSION_DIRECTORY)/archives/$(GOOS)/$(ARCH_LOWER)/$(GIT_TAG).$(GOOS)-$(ARCH_LOWER).tar.gz .
 
+.PHONY: generate-golang-checksum
+generate-golang-checksum: generate-rpm-checksum generate-archive-checksum
+
+.PHONY: generate-rpm-checksum
```

Comment:
The directory path `/tmp/go-extracted/usr/lib/golang` will only contain the extracted contents of the RPM, which are used to produce the tar.gz archive, rather than the RPMs themselves. 

We have two different kinds of RPMs, `arch` and `noarch` RPMs, produced by the build process. 

The `arch`, or architecture-dependent RPMs, are output by the `rpmbuild` process to  `$(VERSION_DIRECTORY)/rpmbuild/$(RPM_ARCH)/golang-*.rpm` (see https://github.com/aws/eks-distro-build-tooling/pull/685/files#diff-5f1e44500fb103dc0ce96c63ccc7dca3b6dbceb62e74976286539988dd20f4a6R108)

The `noarch`, or architecture independent RPMs, are output to `$(VERSION_DIRECTORY)/rpmbuild/noarch/golang-*.rpm` (see https://github.com/aws/eks-distro-build-tooling/pull/685/files#diff-5f1e44500fb103dc0ce96c63ccc7dca3b6dbceb62e74976286539988dd20f4a6R112)

There are also several different RPMs in `noarch` and `arch`:

`arch` RPMs:
- `golang`
- `golang-bin`
- `golang-shared`
- `golang-race`

`noarch` rpms:
- `golang-docs`
- `golang-misc`
- `golang-tests`
- `golang-src`

So, we'll want to generate a checksum for each of these different RPMs.

Reply:
for example, here are where the RPMs for a build of `1.16.15` would end up

arch RPMs:
```
eks-distro-build-tooling/projects/golang/go/1.16/rpmbuild/x86_64/golang-1.16.15-1.amzn2.eks.x86_64.rpm
eks-distro-build-tooling/projects/golang/go/1.16/rpmbuild/x86_64/golang-bin-1.16.15-1.amzn2.eks.x86_64.rpm
eks-distro-build-tooling/projects/golang/go/1.16/rpmbuild/x86_64/golang-shared-1.16.15-1.amzn2.eks.x86_64.rpm
eks-distro-build-tooling/projects/golang/go/1.16/rpmbuild/x86_64/golang-race-1.16.15-1.amzn2.eks.x86_64.rpm
```

noarch RPMs:
```
eks-distro-build-tooling/projects/golang/go/1.16/rpmbuild/noarch/golang-docs-1.16.15-1.amzn2.eks.noarch.rpm
eks-distro-build-tooling/projects/golang/go/1.16/rpmbuild/noarch/golang-misc-1.16.15-1.amzn2.eks.noarch.rpm
eks-distro-build-tooling/projects/golang/go/1.16/rpmbuild/noarch/golang-tests-1.16.15-1.amzn2.eks.noarch.rpm
eks-distro-build-tooling/projects/golang/go/1.16/rpmbuild/noarch/golang-src-1.16.15-1.amzn2.eks.noarch.rpm
```